mesoshttp/client.py

```diff
@@ -713,3 +713,67 @@ def __register(self):
                 if line[count_bytes:]:
                     count_bytes = int(line[count_bytes:])
         return True
+
+    def combine_offers(self, offers, operations, options=None):
+        '''
+        Accept offers with task operations
+
+        :param offers: offers to be accepted
+        :type offers: list
+        :param operations: JSON TaskInfo instances to accept
+        :type operations: list of json TaskInfo
+        :param options: optional filters
+        :type options: JSON filters instances
+        '''
+        if not operations:
+            self.logger.debug('Mesos:Accept:no operation to accept')
+            return True
+
+        if not offers:
+            self.logger.debug('Mesos:Accept:no offers to accept')
+            return True
+
+        ids = [o.get_offer()['id']['value'] for o in offers]
+        offer_ids = [{'value': oid} for oid in ids]
+        self.logger.debug('Mesos:COMBINE Offer ids:' + ','.join(ids))
+
+        headers = {
+            'Content-Type': 'application/json',
+            'Accept': 'application/json',
+            'Mesos-Stream-Id': self.streamId
+        }
+
+        tasks = []
+        for operation in operations:
+            if 'agent_id' not in operation:
+                operation['agent_id'] = {
+                    'value': ids[0]
```

Comment:
You changed operation agent id to ids[0] but this refers to an offer id, not an agent id.
And each operation may have a different agent id if i am correct, forcing to first offer may not be good depending on how mesos manage it.

Merging multiple offers is not so easy as it seems ;-)

Reply:
Oops, you are right about the ids variable.

Concerning the other, the [Scheduler API doc](https://mesos.apache.org/documentation/latest/scheduler-http-api/#accept) is not very clear about whether it is possible to combine offers of different agents. However, in my case I only need to combine offers of the same agent.

I think a better approach would be to leave each operation intact (as it is passed) and let Mesos to process the request and complain if its invalid. Therefore I would remove the whole for loop.
What do you think?